extensions/pkg/controller/worker/genericactuator/actuator_reconcile.go

```diff
@@ -418,12 +415,8 @@ func (a *genericActuator) waitUntilWantedMachineDeploymentsAvailable(ctx context
 				return retryutils.Ok()
 			}
 
-			// update worker status with condition that indicates an ongoing rolling update operation
-			if !workerStatusUpdatedForRollingUpdate {
-				if err := a.updateWorkerStatusMachineDeployments(ctx, worker, extensionsworker.MachineDeployments{}, true); err != nil {
-					return retryutils.SevereError(fmt.Errorf("failed to update the machine status rolling update condition: %w", err))
-				}
-				workerStatusUpdatedForRollingUpdate = true
+			if err := a.updateWorkerStatusMachineDeployments(ctx, worker, extensionsworker.MachineDeployments{}); err != nil {
```

Comment:
Hmm the updateWorkerStatusMachineDeployments here is called with an empty machineDeployments array: `extensionsworker.MachineDeployments{}` so I think it will just produce an empty patch. 
It is also called a bit later in the reconciliation: https://github.com/gardener/gardener/blob/d442fc8ba02d657af07ead21153bbdfcaaa15c99/extensions/pkg/controller/worker/genericactuator/actuator_reconcile.go#L208 so it will be updated there.

Reply:
Very nice catch! We removed this `updateWorkerStatusMachineDeployments` call as it is no-op currently.